### ISSUE TITLE
Enum introspection regression

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#5187]
**Describe the bug**
```python
from enum import Enum


class Foo(Enum):
    x = 1
    y = 2


def contains_one(e: type[Enum]) -> bool:
    values = {v.value for v in e}
    reveal_type(values) # set[Never] in 1.1.307>=, set[Any] in 1.1.306
    return 1 in values # False positive type error here, reportUnnecessaryContains


print(contains_one(Foo))
```

**Expected behavior**
Values should be set[Any] instead of set[Never]. It's also a bit strange to me that I see `"Literal[1]" and "Never" have no overlap`. If set[Never] was correct, I'd expect Never to be compatible with every type/comparison and always overlap.

**VS Code extension or command-line**
pyright cli. 1.1.306 -> 1.1.307 is when this code block started to produce an error message.

**Additional context**
Aside, the codebase I primarily work on I introduced pyright to about 2 years ago when it was fully untyped and already large (rough ballpark 100k lines of code). That codebase now has core folders mostly type checked  and have started comparing errors with new versions more. Before this I'd mostly notice errors from pylance usage for files I was working on.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.311, which I just published. It will also be included in a future release of pylance.
